Sources/Client/Client/Client+Events.swift

```diff
@@ -106,46 +140,47 @@ extension Client {
     }
     
     func subscribeToUnreadCount(for channel: Channel, _ callback: @escaping Completion<ChannelUnreadCount>) -> Cancellable {
-        let subscriptions = SubscriptionBag()
-        
-        let query = ChannelQuery(channel: channel, messagesPagination: [.limit(100)], options: [.state, .watch])
+        let subscription = subscribe(forEvents: [.messageNew, .messageRead], cid: channel.cid) { _ in
+            callback(.success(channel.unreadCount))
+        }
         
-        let urlSessionTask = queryChannel(query: query) { [unowned self] result in
-            if let error = result.error {
-                callback(.failure(error))
-            }
+        // Check if the channel is watching by the client.
+        guard isWatching(channel: channel) else {
+            logger?.log("⚠️ You are trying to subscribe for a channel unread count: (\(channel.cid)), "
+                + "but you didn't start watching it. Please, make a query first with "
+                + "pagination: `[.limit(100)]` and query options: `[.watch, .state]`")
```

Comment:
With the latest backend changes we can show up to 254 unread messages, right?

Reply:
Yes, we can and I think it always was available this magic number :) It was recommended by backend to use 100.